File: main/organizing-containers-of-balls/organizing-containers-of-balls.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

internal static class Solution {
    private static int ReadInt32() => int.Parse(Console.ReadLine());
    
    private static long[] ReadRow(int n)
    {
        var row = Array.ConvertAll(Console.ReadLine().Split(), long.Parse);
        Trace.Assert(row.Length == n);
        return row;
    }
    
    private static long[][] ReadMatrix()
    {
        var n = ReadInt32();
        return Enumerable.Range(0, n).Select(_ => ReadRow(n)).ToArray();
    }
    
    private static IEnumerable<long> RowSums(this long[][] self)
        => self.Select(Enumerable.Sum);
    
    private static IEnumerable<long> ColSums(this long[][] self)
    {
        for (var i = 0; i != self.Length; ++i)
            yield return self.Select(row => row[i]).Sum();
    }
    
    private static void Main()
    {
        for (var t = ReadInt32(); t > 0; --t) {
            var matrix = ReadMatrix();
            
            // We need the bag of container sizes (number of balls in each
            // container) to be equal to the bag of color counts (number of
            // balls of each color).
            var container_sizes = matrix.RowSums().OrderBy(x => x);
            var color_counts = matrix.ColSums().OrderBy(x => x);
            var possible = container_sizes.SequenceEqual(color_counts);
            
            Console.WriteLine(possible ? "Possible" : "Impossible");
        }
    }
}

File: main/organizing-containers-of-balls/organizing-containers-of-balls.py
#!/usr/bin/env python3

def read_val():
    return int(input())

def read_matrix():
    n = read_val()
    
    def read_row():
        row = list(map(int, input().split()))
        if len(row) != n:
            raise ValueError('wrong row size')
        return row
    
    return [read_row() for _ in range(n)]

def row_sums(a):
    return map(sum, a)

def col_sums(a):
    for i in range(len(a)):
        yield sum(row[i] for row in a)

for _ in range(read_val()):
    # We need the bag of container sizes (number of balls in each container)
    # to be equal to the bag of color counts (number of balls of each color).
    a = read_matrix()
    container_sizes = sorted(row_sums(a))
    color_counts = sorted(col_sums(a))
    print('Possible' if container_sizes == color_counts else 'Impossible')
